server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java

```diff
@@ -751,7 +750,7 @@ public void testComputeConsideringShardSizes() {
 
             metadataBuilder.put(
                 IndexMetadata.builder(indexName)
-                    .settings(indexSettings(IndexVersion.current(), 1, 1).put("index.routing.allocation.exclude._name", "node-2"))
+                    .settings(indexSettings(IndexVersion.current(), 1, 1).put("index.routing.allocation.exclude._id", "node-2"))
```

Comment:
So `newNode` uses the argument as a node ID rather than node name?

Reply:
Correct

Reply:
I extracted the cleanups into separate pr: https://github.com/elastic/elasticsearch/pull/102278/files

I would like to use that in other fixes while this stays open for discussions.